components/director/internal/domain/label/query.go

```diff
@@ -0,0 +1,57 @@
+package label
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/lib/pq"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
+)
+
+// QueryFor type defines possible types for quering
+type QueryFor string
+
+const (
+	// QueryForApplication defines the filter query for applications
+	QueryForApplication QueryFor = "app_id"
```

Comment:
It would be nice to have ability to query for  whole`Application` or `Runtime` object

Reply:
But this does not query the database, it just provides you the adequate subquery that can be then used in your context. As an example, you can refer to the runtime repository List func. In this func, the query produces runtimes entities filtered by the subquery provided by the label filter.